test/widgets/action_sheet_test.dart

```diff
@@ -1085,6 +1086,99 @@ void main() {
       });
     });
   });
+  group('channel action sheet', () {
+    late ZulipStream someChannel;
+    late PerAccountStore store;
+
+    Future<void> setupToChannelActionSheet(WidgetTester tester, {
+      required ZulipStream channel,
+      required UnreadMessagesSnapshot unreadMsgs,
+    }) async {
+      someChannel = channel;
+      addTearDown(testBinding.reset);
+
+      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+        realmUsers: [eg.selfUser, eg.otherUser],
+        streams: [someChannel],
+        subscriptions: [eg.subscription(someChannel)],
+        unreadMsgs: unreadMsgs));
+      store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+      connection = store.connection as FakeApiConnection;
+    }
+
+    Future<void> showFromInbox(WidgetTester tester) async {
+      await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+        child: const HomePage()));
+      await tester.pumpAndSettle();
+      check(find.byType(InboxPageBody)).findsOne();
+
+      await tester.pump();
+      await tester.longPress(find.text(someChannel.name).hitTestable());
+      await tester.pump(const Duration(milliseconds: 250));
+    }
+
+    Future<void> showFromSubscriptionList(WidgetTester tester) async {
+      await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+        child: const SubscriptionListPageBody()));
+      await tester.pumpAndSettle();
+      check(find.byType(SubscriptionListPageBody)).findsOne();
+
+      await tester.pump();
+      await tester.longPress(find.text(someChannel.name).hitTestable());
+      await tester.pump(const Duration(milliseconds: 250));
+    }
+
+    final actionSheetFinder = find.byType(BottomSheet);
+    Finder findButtonForLabel(String label) =>
+      find.descendant(of: actionSheetFinder, matching: find.text(label));
+
+    void checkActionSheet() {
+      check(actionSheetFinder).findsOne();
+    }
+
+    void checkButton(String label) {
+      check(findButtonForLabel(label)).findsOne();
+    }
+
+    group('showChannelActionSheet', () {
+      Future<void> prepare(WidgetTester tester, {UnreadMessagesSnapshot? unreadMsgs}) async {
```

Comment:
Instead of `prepare` and `setupToChannelActionSheet`, let's just have one function, `prepare`, following the pattern of the 'topic action sheet' group. The `prepare` function can go outside this `showChannelActionSheet` group, so just in the 'channel action sheet' group.